### PR TITLE
CODON migration changes.

### PR DESCRIPTION
**Pull request by mberacochea**
Small changes to adjust the pipeline infrastructure to CODON

### CODE REVIEW

File: tools/Assembly/antismash/chunking_antismash_with_conditionals/antismash/run_antismash_short.sh [hunk 1]
```diff
@@ -1,12 +1,23 @@
...
-while getopts i:o:g:n:f: option; do
+set -e
+
+while getopts i:o: option; do
 	case "${option}" in
-		i) INPUT=${OPTARG};;
-		o) OUTFOLDER=${OPTARG};;
+	i) INPUT=${OPTARG} ;;
+	o) OUTFOLDER=${OPTARG} ;;
+	*) echo "Invalid option: ${option}" ;;
```
Comment (KateSakharova) [inline tools/Assembly/antismash/chunking_antismash_with_conditionals/antismash/run_antismash_short.sh:9]: I guess we have more bash scripts without *) option...

File: collect_scripts.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+]
+
+# binaries or scripts with no extension
+SPECIFIC_FILES = [
+    "add_header",
+    "awk_tool",
+    "owltools",
+    "faselector",
+    "format_bedfile",
+    "trimmomatic",
```
Comment (KateSakharova) [inline collect_scripts.py:42]: I think we should add '.sh' extensions to add_header, awk_tool, format_bedfile and trimmomatic

File: collect_scripts.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+FILE_EXTENSIONS = [".py", ".sh", ".json", ".pl", ".obo"]
+
+IGNORE_EXTENSIONS = [
+    ".cwl",
+]
+
+IGNORE_LIST = [
+    "__init__.py",
+    "__init__",
+    "copy_utils.py",
```
Comment (KateSakharova) [inline collect_scripts.py:32]: We need to review all scripts in /hps/nobackup2/production/metagenomics/pipeline/tools-v5/utils folder. Many of them are outdated and useless. We can do it after migration (just remove from CODON)

**Review by KateSakharova — APPROVED** [review]
Thank you, Martin, for structuring all these scripts! That makes life easier

File: tools/Assembly/antismash/chunking_antismash_with_conditionals/antismash/run_antismash_short.sh (hunk 1 above)
Comment (mberacochea) [reply tools/Assembly/antismash/chunking_antismash_with_conditionals/antismash/run_antismash_short.sh:9]: yeah, I'll fix them progressively

File: collect_scripts.py (hunk 3 above)
Comment (mberacochea) [reply collect_scripts.py:32]: That is the idea of this script, we will just collect the scripts from the pipeline on deployment. Any changes on that folder will be discarded on each new Jenkins release.

File: collect_scripts.py (hunk 2 above)
Comment (mberacochea) [reply collect_scripts.py:42]: Yeah, I'll do that :).

**Review by caballero — APPROVED** [review]
Changes are appropriate, clean-up is fine, and adding error exceptions for BASH options is a must, however, the antismash script could be a CWL tool but it is not a priority.

**Review by caballero — APPROVED** [review]
changes are ok, porting python v2 to v3 should be a separate task in pipeline 5.1

**Pull request merged**